urlf91495fd27_Rembau_complie_tgz-pJ8-matcher_MatchJ8/src/ll/LL.java

```diff
@@ -0,0 +1,164 @@
+package ll;
+
+import java.util.Hashtable;
+import java.util.Stack;
+import java.util.TreeSet;
+
+public class LL {
+	AnalysisTable at;
+	String str[];
+	TreeSet<String> notEndChars = new TreeSet<String>();
+	public LL(AnalysisTable at,String inStr[]){
+		this.at=at;
+		notEndChars=at.list;
+		//System.out.println("非终结符列表："+at.list);
+		this.str=inStr;
+	}
+	public void handle() throws Exception{
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.